### ISSUE TITLE
Mobile widgets don't show in Outline palette for this file

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#917]
@jhpedemonte

There might be multiple bugs here. I created this file in Maqetta, then saved, and then tried to reopen. Upon reopening, the outline palette doesn't show any widgets. There might be multiple bugs.
- I'm seeing an error in DataListHelper referencing through a null pointer (and therefore assigning to Bill)
- Also, I'm seeing an exception in SpinWheel\* (therefore cc'ing Javier)

If I have time, I'll try to isolate the individual problems.

```
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<title>Untitled</title>
<style type="text/css">
@import "themes/claro/claro.css";
@import "app.css";
@import "./lib/dojo/dojo/resources/dojo.css";
@import "./lib/dojo/dojox/mobile/themes/SpinWheel.css";
</style>
<script type="text/javascript" djConfig="parseOnLoad: true" src="./lib/dojo/dojo/dojo.js"></script>
<script type="text/javascript">
dojo.require('dojox.mobile');
dojo.require('dojox.mobile.parser');
dojo.require('dojox.mobile.deviceTheme');
dojo.require('dojox.mobile.compat');
dojo.require('dojox.mobile.ScrollableView');
dojo.require('dojox.mobile.SwapView');
dojo.require('dojo.data.ItemFileReadStore');
dojo.require('dojox.mobile.RoundRectDataList');
dojo.require('dojox.mobile.EdgeToEdgeDataList');
dojo.require('dojox.mobile.Button');
dojo.require('dojox.mobile.ToggleButton');
dojo.require('dojox.mobile.CheckBox');
dojo.require('dojox.mobile.RadioButton');
dojo.require('dojox.mobile.TextBox');
dojo.require('dijit.form.DataList');
dojo.require('dojox.mobile.ComboBox');
dojo.require('dojox.mobile.TextArea');
dojo.require('dojox.mobile.ExpandingTextArea');
dojo.require('dojox.mobile.TabBar');
dojo.require('dojox.mobile.TabBarButton');
dojo.require('dojox.mobile.Slider');
dojo.require('dojox.mobile.ContentPane');
dojo.require('dojox.mobile.IconContainer');
dojo.require('dojox.mobile.IconItem');
dojo.require('dojox.mobile.SpinWheel');
dojo.require('dojox.mobile.SpinWheelDatePicker');
dojo.require('dojox.mobile.SpinWheelTimePicker');
dojo.require('dojox.mobile.FixedSplitter');
dojo.require('dojox.mobile.FixedSplitterPane');
dojo.require('dojox.mobile.Overlay');</script>
<script src="maqetta/States.js"></script>
<script src="maqetta/maqetta.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"></meta>
</head>
<body class="claro" data-davinci-ws="collapse" dvFlowLayout="true" data-maqetta-device="ipad" id="myapp">
 <select dojoType="dijit.form.DataList" id="DataList_1" data-dojo-props="id:&quot;DataList_1&quot;" dojotype="dijit.form.DataList">
   <option dvwidget="html.option" value="Item 1">
     Item 1</option>
   <option dvwidget="html.option" value="Item 2">
     Item 2</option>
   <option dvwidget="html.option" value="Item 3">
     Item 3</option>
 </select>
 <span dojoType="dojo.data.ItemFileReadStore" id="ItemFileReadStore_2" jsId="ItemFileReadStore_2" data="{&quot;identifier&quot;:&quot;label&quot;,&quot;items&quot;:[{&quot;label&quot;:&quot;Label 1&quot;,&quot;moveTo&quot;:&quot;transitionToPage&quot;},{&quot;label&quot;:&quot;Label 2&quot;,&quot;moveTo&quot;:&quot;transitionToPage&quot;}]}"></span><span dojoType="dojo.data.ItemFileReadStore" id="ItemFileReadStore_1" jsId="ItemFileReadStore_1" data="{&quot;identifier&quot;:&quot;label&quot;,&quot;items&quot;:[{&quot;label&quot;:&quot;Label 1&quot;,&quot;moveTo&quot;:&quot;transitionToPage&quot;},{&quot;label&quot;:&quot;Label 2&quot;,&quot;moveTo&quot;:&quot;transitionToPage&quot;}]}"></span><div id="View_0" dojoType="dojox.mobile.View" style="height: 100%;" selected="true">
<div dojoType="dojox.mobile.SpinWheelDatePicker"></div>
<div dojoType="dojox.mobile.SpinWheelTimePicker"></div>
<div dojoType="dojox.mobile.FixedSplitter" orientation="H">
<div style="width: 200px; border-right: 1px solid black;">
<div dojoType="dojox.mobile.Overlay"></div>
</div>
<div></div>
<div dojoType="dojox.mobile.FixedSplitterPane"></div>
</div>
</div>
 <div id="ScrollableView_0" dojoType="dojox.mobile.ScrollableView" style="display: none; height: 768px;">
  <div label="EdgeToEdgeCategory" dojoType="dojox.mobile.EdgeToEdgeCategory"></div>
  <ul dojoType="dojox.mobile.EdgeToEdgeList">
  <li label="Item 1" dojoType="dojox.mobile.ListItem" sync="true"></li>
    <li label="Item 2" dojoType="dojox.mobile.ListItem" sync="true"></li>
    <li label="ListItem" dojoType="dojox.mobile.ListItem" sync="true"></li>
  </ul>
  <ul dojoType="dojox.mobile.EdgeToEdgeDataList" store="ItemFileReadStore_2" query="{&quot;label&quot;:&quot;*&quot;}"></ul>
  <button label="Button" dojoType="dojox.mobile.Button"></button>
  <button label="ToggleButton" dojoType="dojox.mobile.ToggleButton"></button>
  <div dojoType="dojox.mobile.Switch"></div>
  <input type="checkbox" dojoType="dojox.mobile.CheckBox"></input>
  <input type="radio" dojoType="dojox.mobile.RadioButton"></input>
  <input type="text" dojoType="dojox.mobile.TextBox"></input>
  <input type="text" dojoType="dojox.mobile.ComboBox" data-dojo-props="value:&quot;Item 1&quot;, list:&quot;DataList_1&quot;"></input>
  <textarea dojoType="dojox.mobile.TextArea"></textarea>
  <textarea dojoType="dojox.mobile.ExpandingTextArea"></textarea>
  <ul dojoType="dojox.mobile.TabBar" barType="segmentedControl">
    <li label="TabBarButton" dojoType="dojox.mobile.TabBarButton" sync="true" selected="true"></li>
    <li label="TabBarButton" dojoType="dojox.mobile.TabBarButton" sync="true"></li>
    <li label="TabBarButton" dojoType="dojox.mobile.TabBarButton" sync="true"></li>
    <li label="TabBarButton" dojoType="dojox.mobile.TabBarButton" sync="true"></li>
  </ul>
  <input type="range" dojoType="dojox.mobile.Slider"></input>
  <div dojoType="dojox.mobile.ContentPane"></div>
  <ul dojoType="dojox.mobile.IconContainer" iconBase="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB0AAAAdCAMAAABhTZc9AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAMBQTFRFkaO4nazAucXThpmyt8PRsr7O3+Tr2N7mw83ZdIqmmqq+g5evq7jJ4ufsy9PegZWuztbgpLLEq7nJiZy0d4yomKi94ebsfpOsfJGrlqe8jp+3wsvYobHD6OzwboWiiJuzfZKsrrvMprTGzdXfx9DbydLdxc7au8bUb4ajxs/bc4mlk6S6197lwcvX2d/neY6pv8nW5enuo7LE4OXrn67Bmam+1dzkz9bgyNDbqLbHcYekjqC2lKW62+Hoipy0////GCC5wQAAAEB0Uk5T////////////////////////////////////////////////////////////////////////////////////AMJ7sUQAAAFASURBVHjafNPZcoIwFIDhExAULKAYBJfixq4stRZKEHn/tyoI6Y1Nv4tM5p85k8nFgaZJ5vB4BfOkaaDx5Un9/qqeyH4DY0P5+JtiJGB4nywTGfi8Zsl50HI2DcyCzQTrm82CNKM24tLNMrWe/ZYU8IhyFUXkYkVZcrRgkA6UI4qiX7SHQIsEUYWOHf0otJ8QiroujgMUAZGwtjjL8cW9ZhcV+Ze8ROUNyeeFhgkEpIPd04xbG+f7fbtaOTPOwl0NwO4JnrexbaIFtn14Xp+GWTKNQ4GgTZitiReGhz4G7bs9XKW601kJDrfvG4Eopebuk/PgaYlA2g8qdcDRspcATwf6ieJpwpDuBrJ3ioVy5KlvtOxSsCo2C8w7mwnaF5sGPLqxIB4MdGVBMoyhZKmSdhdgW+qvyi20u/DvHv0IMAAbv4T3o6uF6wAAAABJRU5ErkJggg%3D%3D">
    <li label="Item 1" dojoType="dojox.mobile.IconItem"></li>
    <li label="Item 2" dojoType="dojox.mobile.IconItem"></li>
    <li label="IconItem" dojoType="dojox.mobile.IconItem"></li>
  </ul>
  <div dojoType="dojox.mobile.SpinWheel">
    <div dojoType="dojox.mobile.SpinWheelSlot" labels="[&quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;,&quot;E&quot;]"></div>
    <div dojoType="dojox.mobile.SpinWheelSlot" labels="[&quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;,&quot;E&quot;]"></div>
  </div>
</div>
 <div id="SwapView_0" dojoType="dojox.mobile.SwapView" style="display: none; height: 768px;">
  <h1 label="Heading" dojoType="dojox.mobile.Heading">
   <div label="ToolBarButton" dojoType="dojox.mobile.ToolBarButton" sync="true"></div>
 </h1>
  <div dojoType="dojox.mobile.RoundRect"></div>
  <div label="RoundRectCategory" dojoType="dojox.mobile.RoundRectCategory"></div>
  <ul dojoType="dojox.mobile.RoundRectList">
  <li label="Item 1" dojoType="dojox.mobile.ListItem" sync="true"></li>
    <li label="Item 2" dojoType="dojox.mobile.ListItem" sync="true"></li>
  </ul>
  <ul dojoType="dojox.mobile.RoundRectDataList" store="ItemFileReadStore_1" query="{&quot;label&quot;:&quot;*&quot;}"></ul>
</div>
</body>
</html>
```

**Comment by billreed63** [comment]
I have checked in fixes for the helper exceptions.. but I am still seeing the excception in dojox.mobile.SpinWheelSlot

**Comment by jhpedemonte** [comment]
The SpinWheel error happens because it doesn't like to be hidden.  Easy way to test:
- Add **View**.
- Add **SpinWheel**.
- In Outline, click on eyeball next to SpinWheel.

Preview or close/reopen file and you'll see the exception.

**Comment by jhpedemonte** [comment]
Looks like a Dojo bug to me.  If the SpinWheel is visible, then in `SpinWheelSlot.getCenterPanel()`, the properties `offsetTop`/`offsetHeight` have non-zero values, and the function returns a panel instance.  If SpinWheel is hidden/invisible (for example, in a secondary View that isn't currently selected), then the `offset*` properties are always zero and `getCenterPanel()` returns `null`, causing exception.

Guess I'll have to open a bug on Dojo, then...

@peller

**Comment by jhpedemonte** [comment]
Opened Dojo issue: http://bugs.dojotoolkit.org/ticket/13871

**Comment by jhpedemonte** [comment]
SpinWheel error fixed in latest Dojo 1.7 release.
